opengrok-indexer/src/test/java/org/opengrok/indexer/index/IndexerTest.java

```diff
@@ -451,39 +452,32 @@ public void testIncrementalIndexAddRemoveFile() throws Exception {
      * @throws Exception
      */
     @Test
+    @EnabledIf("mkinfoInPath")
     public void testBug11896() throws Exception {
+        RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+        env.setSourceRoot(repository.getSourceRoot());
+        env.setDataRoot(repository.getDataRoot());
+        Executor executor;
 
-        boolean test = true;
-        if (FileUtilities.findProgInPath("mkfifo") == null) {
-            System.out.println("Error: mkfifo not found in PATH !\n");
-            test = false;
-        }
-
-        if (test) {
-            RuntimeEnvironment env = RuntimeEnvironment.getInstance();
-            env.setSourceRoot(repository.getSourceRoot());
-            env.setDataRoot(repository.getDataRoot());
-            Executor executor;
-
-            executor = new Executor(new String[] {"mkdir", "-p", repository.getSourceRoot() + "/testBug11896"});
-            executor.exec(true);
+        executor = new Executor(new String[] {"mkdir", "-p", repository.getSourceRoot() + "/testBug11896"});
+        executor.exec(true);
 
-            executor = new Executor(new String[] {"mkfifo", repository.getSourceRoot() + "/testBug11896/FIFO"});
-            executor.exec(true);
+        executor = new Executor(new String[] {"mkfifo", repository.getSourceRoot() + "/testBug11896/FIFO"});
+        executor.exec(true);
 
-            Project project = new Project("testBug11896");
-            project.setPath("/testBug11896");
-            IndexDatabase idb = new IndexDatabase(project);
-            assertNotNull(idb);
-            MyIndexChangeListener listener = new MyIndexChangeListener();
-            idb.addIndexChangedListener(listener);
-            System.out.println("Trying to index a special file - FIFO in this case.");
-            idb.update();
-            assertEquals(0, listener.files.size());
+        Project project = new Project("testBug11896");
+        project.setPath("/testBug11896");
+        IndexDatabase idb = new IndexDatabase(project);
+        assertNotNull(idb);
+        MyIndexChangeListener listener = new MyIndexChangeListener();
+        idb.addIndexChangedListener(listener);
+        System.out.println("Trying to index a special file - FIFO in this case.");
+        idb.update();
+        assertEquals(0, listener.files.size());
+    }
 
-        } else {
-            System.out.println("Skipping test for bug 11896. Could not find a mkfifo in path.");
-        }
+    public boolean mkinfoInPath() {
```

Comment:
No need for `public` modifier, can be even `private` it seems

Reply:
https://junit.org/junit5/docs/snapshot/user-guide/index.html#writing-tests-conditional-execution-custom has an example with package private access so I think we should go with that :)